dcp_prototype/backend/wrangling/migrations/scripts/metadata_aggregator.py

```diff
@@ -0,0 +1,166 @@
+import argparse
+import tarfile
+from urllib.parse import urlparse
+import boto3
+from botocore.exceptions import ClientError
+
+import threading
+import queue
+import os
+import tempfile
+import sys
+
+"""This is a script to download all flattened DCP-1.0 metadata files from S3.
+The files can then be tarred and gzipped, then copied back to S3."""
+
+
+def download_file(prefix, bucket, filequeue, download_dir):
+    session = boto3.session.Session()
+    s3_client = session.client("s3")
+    while True:
+        filename = filequeue.get()
+
+        # check for more files to process
+        if filename is None:
+            break
+
+        object_name = prefix + "/" + filename
+        qsize = filequeue.qsize()
+        if qsize % 1000 == 0 and qsize > 0:
+            print(f"Working with JSON input file: {object_name}, queued size={qsize}")
+
+        download_dest = os.path.join(download_dir, filename)
+        try:
+            s3_client.download_file(bucket, object_name, download_dest)
+        except Exception as e:
+            print(f"Failed {str(e)} : {bucket} {object_name} {download_dest}")
+
+        filequeue.task_done()
+
+
+def download_files(s3_uri, num_threads, dirname):
+
+    bucket_name = urlparse(s3_uri).netloc
+    prefix = urlparse(s3_uri).path
+    if prefix.startswith("/"):
+        prefix = prefix[1:]
+
+    client = boto3.client("s3")
+    paginator = client.get_paginator("list_objects")
+    page_iterator = paginator.paginate(Bucket=bucket_name, Prefix=prefix)
+
+    filequeue = queue.Queue()
+    nfiles = 0
+    prefixparts = len(prefix.split("/"))
+    for page in page_iterator:
+        bucket_objects = page.get("Contents")
+        for object in bucket_objects:
+            object_filename = object.get("Key")
+            if "data_files" in object_filename:
+                continue
+            parts = object_filename.split("/")
+            if len(parts) == prefixparts + 1:
```

Comment:
What is this checking for?

Reply:
This is making sure only the top level objects in the bucket are downloaded.

Reply:
Got it thanks.